doc/source/testing_pull_requests.rst

```diff
@@ -0,0 +1,178 @@
+      file to build it so we must tell python-for-android what we want via
+      arguments
+    - be sure to at least edit the following arguments when running the above
+      command, since the default set in there it's unlikely that match your
+      installation:
+
+        - `--ndk-dir`: An absolute path to your android's NDK dir
+        - `--sdk-dir`: An absolute path to your android's SDK dir
+        - `--debug`: this one enables the debug mode of python-for-android,
+          which will show all log messages of the build. You can omit this one
+          but it's worth it to be mentioned, since this it's useful to us when
+          trying to find the source of the problem when things goes wrong
+    - The apk generated by the above command should be located at the root of
+      of the cloned repository, were you run the command to build the apk
+    - The testapps distributed with python-for-android are located at
+      `testapps` folder under the main folder project
+    - All the builds of python-for-android are located at
+      `~/.local/share/python-for-android`
+    - You should have a downloaded copy of the android's NDK and SDK
+
+Installing python-for-android using the github's branch of the pull request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Enter inside the directory of the cloned repository mentioned in
+  `Common steps` and install it via pip, eg.::
+
+    .. codeblock:: bash
+
+        cd p4a-feature-fix-numpy
+        pip3 install . --upgrade --user
+
+- Now, go inside the `testapps` directory (we assume that you still are inside
+  the cloned repository)::
+
+    .. codeblock:: bash
+
+        cd testapps
+
+- Run the build of the apk via the freshly installed copy of python-for-android
+  by running a similar command than below::
+
+    .. code-block:: bash
+
+        python3 setup_testapp_python3_sqlite_openssl.py apk \
+            --ndk-dir=/media/DEVEL/Android/android-ndk-r20 \
+            --sdk-dir=/media/DEVEL/Android/android-sdk-linux \
+            --android-api=27 \
+            --arch=arm64-v8a \
+            --debug
+
+
+Things that you should know:
+
+    - In the example above, we override some variables that are set in
+      `setup_testapp_python3_sqlite_openssl.py`, you could also override them
+      by editing this file
+    - be sure to at least edit the following arguments when running the above
+      command, since the default set in there it's unlikely that match your
+      installation:
+
+        - `--ndk-dir`: An absolute path to your android's NDK dir
+        - `--sdk-dir`: An absolute path to your android's SDK dir
+
+.. warning:: Once you finish the pull request tests remember to go back to the
+             master or develop versions of python-for-android, since you just
+             installed the python-for-android files of the `pull request`
+
+Using buildozer with a custom app
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
I actually often use a method pretty close to that one. The method is terribly wrong so I won't share it in the docs.
I edit: `~/.local/lib/python3.6/site-packages/buildozer/targets/android.py` and replace the clone command by the fork.
BTW I think at some point buildozer should make it possible from the spec file, to not only chose the branch, but also the fork.

Reply:
> BTW I think at some point buildozer should make it possible from the spec file, to not only chose the branch, but also the fork.

Yeah...you are right...this should be implemented in buildozer :+1: